monitoring/bin/deploy_monitoring_cluster.sh

```diff
@@ -65,10 +65,37 @@ fi
 # Elasticsearch Datasource for Grafana
 ELASTICSEARCH_DATASOURCE="${ELASTICSEARCH_DATASOURCE:-false}"
 if [ "$ELASTICSEARCH_DATASOURCE" == "true" ]; then
-  log_verbose "Provisioning Elasticsearch datasource for Grafana"
-  kubectl delete secret -n $MON_NS --ignore-not-found grafana-datasource-es
-  kubectl create secret generic -n $MON_NS grafana-datasource-es --from-file monitoring/grafana-datasource-es.yaml
-  kubectl label secret -n $MON_NS grafana-datasource-es grafana_datasource=1 sas.com/monitoring-base=kube-viya-monitoring
+
+  # Check to see if logging components have already been deployed.
+  if [[ $(kubectl get pods -A -l app.kubernetes.io/instance=v4m-fb -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+    log_verbose "No logging components found.  Skipping Elasticsearch data source set up.";
+  elif [[ $(kubectl get pods -A -l app.kubernetes.io/instance=v4m-fb -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) > 1 ]]; then
+    log_verbose "Too many logging components found.  Skipping Elasticsearch data source set up.";
```

Comment:
This should never happen....and (I think/hope) should be difficult (if not impossible) to do.  Was this added just to be extra careful or am I wrong about how likely/possible this is?

Reply:
Extra careful...Do you know what happens if you deploy logging to two different namespaces?

Reply:
Hmmmm, good question.  Just tried it: deployment script fails when it goes to deploy ODFE...Helm sees there are existing resources with the same name and aborts.  Hmmm, we may need/want to add logic to check for a pre-existing deployment in a different namespace to the driver script.  I'll put it on the "to do" list.